docs/source/authentication_with_modelon_impact.rst

```diff
@@ -4,16 +4,10 @@ Authentication with Modelon Impact
 This page explains how to authenticate and login to the Modelon Impact server from the Python client 
 in various environments.
 
-The cloud/on-premise architecture for Modelon Impact uses JupyterHub as one of its components. 
-As a consequence, when connecting via an external URL like https://impact.modelon.cloud, 
-JupyterHub gateway authentication is necessary. From a notebook that is active within the MI 
-installation, one can create a connection to the external URL. In that case, this will be automatic
-based on the automatically populated ``JUPYTERHUB_API_TOKEN`` environment variable.
-
-An API key must be used in addition to the ``JUPYTERHUB API TOKEN`` to authenticate against the Modelon 
+The cloud/on-premise architecture for Modelon Impact uses an API key to authenticate against the Modelon 
 Impact server operating inside the JupyterHub environment. If the client library is used within the
-JupyterHub environment, this is likewise handled automatically based on the key stored at 
-``~/.impact/api.key`` in that environment.
+JupyterHub environment, this is likewise handled automatically based on the key stored in the environmental
+variable ``MODELON_IMPACT_CLIENT_API_KEY``.

```

Comment:
Some rewrite and may need to iterate:

In order to work with Modelon Impact REST API a non-browser code needs to pass  `impact-api-key: <api key>` header with all requests. The Python `Client()` constructor will set this header using the value of API key from either (in order of priority):
- explicit constructor parameter or interactively given by a user in interactive mode;
- environment variable `MODELON_IMPACT_CLIENT_API_KEY`;
- `$HOME/.impact/api.key` file.

The `MODELON_IMPACT_CLIENT_API_KEY` environment variable is automatically set with a temporary user specific API key inside the Modelon Impact environment. Therefore explicit user input is not necessary when the Python client is used in a script, notebook or custom function running within the Modelon Impact by a logged in user.

When using Python client from outside Modelon Impact, e.g., from a desktop, the API key needs to be generated via user interface.

Reply:
Thanks Iakov for the nice write up. I have now made the suggested changes with some minor tweaks to the order of preference. Please let me know if you find this reasonable.